analyzers/src/SonarAnalyzer.CSharp/Helpers/CSharpDiagnosticAnalyzerContextHelper.cs

```diff
@@ -32,8 +32,9 @@ internal static class CSharpDiagnosticAnalyzerContextHelper
                                                                                params TSyntaxKind[] syntaxKinds) where TSyntaxKind : struct =>
             context.RegisterSyntaxNodeActionInNonGenerated(CSharpGeneratedCodeRecognizer.Instance, action, syntaxKinds);
 
-        public static void RegisterSyntaxNodeActionInNonGenerated<TSyntaxKind>(this CompilationStartAnalysisContext context, Action<SyntaxNodeAnalysisContext> action, params TSyntaxKind[] syntaxKinds)
-            where TSyntaxKind : struct =>
+        public static void RegisterSyntaxNodeActionInNonGenerated<TSyntaxKind>(this SonarCompilationStartAnalysisContext context,
+                                                                               Action<SyntaxNodeAnalysisContext> action,
+                                                                               params TSyntaxKind[] syntaxKinds) where TSyntaxKind : struct =>
             context.RegisterSyntaxNodeActionInNonGenerated(CSharpGeneratedCodeRecognizer.Instance, action, syntaxKinds);
 
         public static void RegisterSyntaxTreeActionInNonGenerated(this SonarAnalysisContext context, Action<SyntaxTreeAnalysisContext> action) =>
```

Comment:
It is indeed unused (also in the latest commit). At the same time, it is part of the registration framework available to all rules. 

Can I throw a UT on it instead? :) In order to have uniformed registration available for all our devs, that might not be brave enough to create methods like this while working on a rule.

Reply:
In my opinion, we should not have unused code. I trust our colleagues to add it or raise a hand if they need help.

If you insist on keeping it, it should be tested indeed. This will have the downside that it will hide the fact that it's not used 😢

Reply:
In this (very rare) case, I'd like to have it ready.  I'll throw an UT on it later, as it will get renamed soon and I'll create Bump coverage PR with the new name